concordium-consensus/src/Concordium/KonsensusV1/Scheduler.hs

```diff
@@ -402,10 +415,75 @@ processBlockRewards ParticipatingBakers{..} TransactionRewardParameters{..} miss
   where
     hasValidatorSuspension = sSupportsValidatorSuspension (sAccountVersionFor (protocolVersion @pv))
 
+-- | Suspend all validators with more missed rounds than the threshold specified
+--  in the chain parameters. Logs the suspension of a validator in a special
+--  transaction outcome.
+processSuspensions ::
+    forall pv m.
+    ( pv ~ MPV m,
+      BlockStateStorage m,
+      IsConsensusV1 pv,
+      PVSupportsValidatorSuspension pv
+    ) =>
+    Timestamp ->
+    UpdatableBlockState m ->
+    m (UpdatableBlockState m)
+processSuspensions timestamp bs0 = do
+    -- get chain paramers
+    ps <- bsoGetChainParameters bs0
+    case _cpValidatorScoreParameters ps of
+        NoParam -> return bs0 -- Nothing to do here.
+        SomeParam (ValidatorScoreParameters{..}) -> do
+            -- get baker pool reward details
+            bprd <- bsoGetBakerPoolRewardDetails bs0
+            -- account indexes that will be suspended
+            let ixs =
+                    [ bakerAccountIndex bid
+                      | (bid, rd) <- Map.toList bprd,
+                        fromCondDef (suspensionPrimed rd) False,
+                        fromCondDef (missedRounds rd) 0 > _vspMaxMissedRounds
+                    ]
```

Comment:
If we reset the `primed` flag when a validator contributes, then we should only need to check that flag here, and not the number of missed rounds. Assuming that we only set the primed flag in the first place when the validator exceeds the number of missed rounds by the payday.

Reply:
this has been changd.